Jellyfin.Api/Controllers/TimeSyncController.cs

```diff
@@ -22,18 +22,14 @@ public class TimeSyncController : BaseJellyfinApiController
         public ActionResult<UtcTimeResponse> GetUtcTime()
         {
             // Important to keep the following line at the beginning
-            var requestReceptionTime = DateTime.UtcNow.ToUniversalTime().ToString("o", DateTimeFormatInfo.InvariantInfo);
+            var requestReceptionTime = DateTime.UtcNow.ToUniversalTime();
 
-            var response = new UtcTimeResponse();
-            response.RequestReceptionTime = requestReceptionTime;
-
-            // Important to keep the following two lines at the end
-            var responseTransmissionTime = DateTime.UtcNow.ToUniversalTime().ToString("o", DateTimeFormatInfo.InvariantInfo);
-            response.ResponseTransmissionTime = responseTransmissionTime;
+            // Important to keep the following line at the end
+            var responseTransmissionTime = DateTime.UtcNow.ToUniversalTime();
 
             // Implementing NTP on such a high level results in this useless
             // information being sent. On the other hand it enables future additions.
-            return response;
+            return new UtcTimeResponse(requestReceptionTime, responseTransmissionTime);
```

Comment:
Exactly, that's what I meant with `high level` in the comment above. Yet, I wanted to maintain both values, even if practically the same, so that client side implementation would still resemble the NTP one.

> We also have a response time middleware now that sets the best estimate for response time in the custom header `X-Response-Time-ms`.

Using this one would definitely make more sense, hoping it's more accurate. Is there also a time for when the request has been received by the server?

Reply:
There's no built-in method for this. Our middleware is the best guesstimation.